### PR TITLE
spellcheck: Sync 2.0-dev with master

### LINKED ISSUE #2897: Add ttRPC to spellcheck
Add **ttRPC** to spell check dictionary to fix architecture update PR that is failing (https://github.com/kata-containers/kata-containers/pull/798).

### PR DESCRIPTION
**Pull request by cmaf**
Update 2.0-dev branch with main.txt and projects.txt from master.

Fixes #2897

Signed-off-by: Chelsea Mafrica <chelsea.e.mafrica@intel.com>

### CODE REVIEW

**Comment by cmaf** [comment]
Backport is #2902, which adds ttRPC to spell check. This PR also syncs master and 2.0-dev.

**Comment by jodh-intel** [comment]
Thanks for raising @cmaf, but I'm seeing a different set of differences:

```diff
diff --git a/cmd/check-spelling/data/main.txt b/cmd/check-spelling/data/main.txt
index dd5e356..d4423f7 100644
--- a/cmd/check-spelling/data/main.txt
+++ b/cmd/check-spelling/data/main.txt
@@ -98,4 +98,3 @@ webhook/AB
 whitespace
 workflow/A
 Xeon/A
-netlink
diff --git a/cmd/check-spelling/data/projects.txt b/cmd/check-spelling/data/projects.txt
index e91a805..4400bc2 100644
--- a/cmd/check-spelling/data/projects.txt
+++ b/cmd/check-spelling/data/projects.txt
@@ -25,12 +25,10 @@ GitHub/B
 GoDoc/B
 golang/B
 Golang/B
-Grafana/B
 Huawei/B
 iPerf/B
 IPerf/B
 Istio/B
-Jaeger/B
 Jenkins/B
 journald/B
 Kata/B
@@ -49,12 +47,10 @@ Minikube/B
 MITRE/B
 Netlify/B
 Nginx/B
-OpenCensus/B
 OpenPGP/B
 OpenShift/B
 OpenSSL/B
 OpenStack/B
-OpenTelemetry/B
 OpenTracing/B
 osbuilder/B
 packagecloud/B
```

I think it would be clearer to add one commit for the missing terms above and one more to add `ttRPC`, but I'm unclear on the other words on this PR.

File: cmd/check-spelling/data/main.txt [hunk 1]
```diff
@@ -78,6 +83,7 @@ teardown
 templating
 timestamp/AB
 tracability
+ttRPC
```
Comment (jodh-intel) [inline cmd/check-spelling/data/main.txt:86]: btw, you probably want to make this `ttRPC/B` as that will match "ttRPC" and also "ttRPC's". These one-letter rules are defined here:

- https://github.com/kata-containers/tests/blob/master/cmd/check-spelling/data/rules.aff
  ("A" is for handling plurals and "B" is for possession [`'s`]).

Comment (cmaf) [reply cmd/check-spelling/data/main.txt]: done

**Comment by cmaf** [comment]
@jodh-intel I updated to have 2 commits, the first with the original sync and the second with ttRPC plus additional words that caused warnings when I ran the spell check locally.

**Comment by cmaf** [comment]
@jodh-intel the commit message was also updated for https://github.com/kata-containers/tests/pull/2903 before it was merged to reflect that additional words from the architecture doc were added

**Review by jodh-intel — APPROVED** [review]

**Review by GabyCT — APPROVED** [review]

**Pull request merged**